yamllint/rules/__init__.py

```diff
@@ -63,8 +65,11 @@
 }
 
 
-def get(id):
-    if id not in _RULES:
+def get(id, rules=None):
+    if rules is None:
+        rules = yamllint.plugins.update_rules_map(_RULES)
```

Comment:
I see 2 issues:
1. It appears that this call *modifies the `_RULES` object.*
2. This `get()` function is called *many* times by the linter, it would be better not to call `yamllint.plugins.update_rules_map()` every time.

What do you think of:
```python
_EXTERNAL_RULES = None


def get(id):
    global _RULES, _EXTERNAL_RULES

    if _EXTERNAL_RULES is None:
        _EXTERNAL_RULES = yamllint.plugins.load_plugin_rules()

    if id in _RULES:
        return _RULES[id]
    elif id in _EXTERNAL_RULES:
        return _EXTERNAL_RULES[id]
    else:
        raise ValueError('no such rule: "%s"' % id)
```
?

Reply:
Absolutely right! I like the code looks best and don't have any objections about it.

Reply:
One thing I forgot to mention.

How about to add arguments {rules=None, external_rules=Noe} (both will fallback to_RULES, _EXTERNAL_RULES if no arguments were given) to 'get' function to make it easy to test?

Reply:
Could you please take a look at the commit 01554f2 ? I think it got better.